pkg/apis/resource/validation/validation.go

```diff
@@ -119,10 +119,41 @@ func validateDeviceClaim(deviceClaim *resource.DeviceClaim, fldPath *field.Path,
 	return allErrs
 }
 
-func gatherRequestNames(deviceClaim *resource.DeviceClaim) sets.Set[string] {
-	requestNames := sets.New[string]()
+type requestNames map[string]sets.Set[string]
+
+func (r requestNames) Has(s string) bool {
+	segments := strings.Split(s, "/")
+	// If there are more than one / in the string, we
+	// know there can't be any match.
+	if len(segments) > 2 {
+		return false
+	}
+	// If the first segment doesn't have a match, we
+	// don't need to check the other one.
+	subRequestNames, found := r[segments[0]]
+	if !found {
+		return false
+	}
+	if len(segments) == 1 {
+		return true
+	}
+	// If the first segment matched and we have another one,
+	// check for a match for that too.
+	return subRequestNames.Has(segments[1])
+}
+
+func gatherRequestNames(deviceClaim *resource.DeviceClaim) requestNames {
+	requestNames := make(requestNames)
 	for _, request := range deviceClaim.Requests {
-		requestNames.Insert(request.Name)
```

Comment:
Oh darn. Did I forget to validate uniqueness?!

Reply:
Strictly speaking, the names don't need to be unique. A config with a certain request name could apply to several different requests with that same name.

But this feels unusual and wasn't intended.

Reply:
Right. I wasn't sure anymore whether it was covered elsewhere, but it is. Therefore it's okay in this code here to not check for existence of the new entry.

Uniqueness certainly was intended. I think it's not documented as requirement for the DeviceRequest.Name and should be added, similar to what I suggested for the new subrequest name. Can you include that in this PR?